### PR TITLE
Implemented autoformat of sources during build

### LINKED ISSUE #9: Automatic code formatter/linter
Currently during PRs we often see code being reformated by IDE - that makes reading PR harder.

We need to have some kind of autoformatter which would do code format on build + committed settings for this code formatter.

### PR DESCRIPTION
**Pull request by DiverOfDark**
resolves #9

### CODE REVIEW

**Review by jMaddrell — APPROVED** [review]

File: Firmware/scripts/run_clangformat.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    print("Executing Formatter")
+
+    check_folders = [env.get("PROJECT_INCLUDE_DIR"), env.get("PROJECT_SRC_DIR")]
+    
+    project_dir = env.get("PROJECT_DIR") + "\\"
+    check_folders_without_prefix = [str(i).replace( project_dir, '') for i in check_folders]
+
+    print("check folders", check_folders_without_prefix);
+    
+    subprocess.run("pip -q install clang-format")
```
Comment (jMaddrell) [inline Firmware/scripts/run_clangformat.py:15]: On macOS python installed via brew doesn't provide a pip binary only pip3, I swapped this to the following locally and it works as expected:
```
env.Execute("\"$PYTHONEXE\" -m pip install clang-format")
```

Comment (DiverOfDark) [reply Firmware/scripts/run_clangformat.py]: Thanks, updated.

Comment (jMaddrell) [reply Firmware/scripts/run_clangformat.py]: With the current change I'm still getting failures during build:
```
FileNotFoundError: [Errno 2] No such file or directory: '"$PYTHONEXE" -m pip -q install clang-format':
```

Comment (DiverOfDark) [reply Firmware/scripts/run_clangformat.py]: my bad, updated once more

**Review by jMaddrell — APPROVED** [review]

File: Firmware/scripts/run_clangformat.py (hunk 1 above)
Comment (jMaddrell) [reply Firmware/scripts/run_clangformat.py]: thanks working now

**Pull request merged**